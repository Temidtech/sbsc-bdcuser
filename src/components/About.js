import React, { PureComponent } from 'react';
import { Text, View, TouchableHighlight, Linking } from 'react-native';
import { ContentWrapper } from '.';
import { AboutStyles as styles } from '../styles';

const aboutText = `
The acronym ABCON depicts Association of\n Bureaux De Change Operators of Nigeria.\n The historical background of ABCON dates back\n to 4th of February, 1997 when its registration\n was approved by the Corporate Affairs\n Commission (C.A.C).\n\n It came into being from a vision of some\n bureaux de change operators numbering more\n than 50(fifty) in an attempt to create an\n environment of sanctity in the business of\n foreign exchange.\n\n Hither-to, the country-Nigeria was experiencing\n an economic recession which emanated from\n the restrictions placed on foreign exchange by a\n government policy named "Structural\n Adjustment Program" (SAP) prevalent in the '80s.
`;
class About extends PureComponent {
  render() {
    return (
      <ContentWrapper>
        <View style={styles.container}>
          <TouchableHighlight style={styles.aboutContent}>
            <Text style={styles.aboutText}>{aboutText}</Text>
          </TouchableHighlight>
          <TouchableHighlight
            onPress={() => Linking.openURL('http://abcon.org.ng/AboutUs.aspx')}
            underlayColor="#19B01D"
            style={styles.readMoreButton}>
            <Text style={styles.cta}>Read more</Text>
          </TouchableHighlight>
        </View>
      </ContentWrapper>
    );
  }
}

export default About;
