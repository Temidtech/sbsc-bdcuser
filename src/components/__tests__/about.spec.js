import React from 'react';
import renderer from 'react-test-renderer';
import About from '../About';

describe('<About />', () => {

  it('renders the About component correctly', () => {
    const tree = renderer.create(
      <About />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
