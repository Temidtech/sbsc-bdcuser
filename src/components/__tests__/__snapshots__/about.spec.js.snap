// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> renders the About component correctly 1`] = `
<View
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/assets/img/backgd.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,.7)",
        "height": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "borderWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "marginRight": 10,
            "marginTop": 10,
            "width": 50,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/img/converter.png",
            }
          }
          style={
            Object {
              "height": 33,
              "width": 25,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "height": "100%",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#105410",
            "borderRadius": 30,
            "height": 450,
            "justifyContent": "center",
            "marginTop": 10,
            "opacity": 0.8,
            "width": "98%",
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 15,
              "padding": 10,
            }
          }
        >
          
The acronym ABCON depicts Association of
 Bureaux De Change Operators of Nigeria.
 The historical background of ABCON dates back
 to 4th of February, 1997 when its registration
 was approved by the Corporate Affairs
 Commission (C.A.C).

 It came into being from a vision of some
 bureaux de change operators numbering more
 than 50(fifty) in an attempt to create an
 environment of sanctity in the business of
 foreign exchange.

 Hither-to, the country-Nigeria was experiencing
 an economic recession which emanated from
 the restrictions placed on foreign exchange by a
 government policy named "Structural
 Adjustment Program" (SAP) prevalent in the '80s.

        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#02a901",
            "borderRadius": 5,
            "height": 50,
            "justifyContent": "center",
            "marginTop": 15,
            "width": "70%",
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontFamily": "OpenSans-SemiBold",
              "fontSize": 15,
            }
          }
        >
          Read more
        </Text>
      </View>
    </View>
  </View>
</View>
`;
